### ISSUE TITLE
Improve `jupytext` / `ipynb` workflows to automate tedium

### ISSUE THREAD

**Issue by HarrisonWilde** [nestauk/ds-cookiecutter#197]
If we are going to enforce `jupytext` why not have some automation around conversion. Is there already a solution for this I just haven't realised? What is the guidance on working with `jupytext` in VSCode, if there is a solid way (I am not aware) then perhaps the conversion point is moot?

**Comment by caldwellst** [comment]
Been a long while since I worked with notebooks, so don't exactly remember the workflow, but believe there were ways to make conversion simple and easy. We also never version controlled `.ipynb` files, just the `.md` or `.py` files.

However, I think we should consider doing away with `jupytext`. Someone showed me when I started here there's now a [GitHub Feature Preview](https://github.blog/changelog/2023-03-01-feature-preview-rich-jupyter-notebook-diffs/) for rich diffs of Jupyter Notebooks. You have to activate for your account here.

![image](https://github.com/user-attachments/assets/4de659ce-01c0-459f-97b8-2be40ff4dd7a)

If the only problem we are solving for is better ability to review changes in VC, seems like this has been solved on GitHub's end!

**Comment by HarrisonWilde** [comment]
Yes this is a great solution and much simpler, I notice a lot of the projects don't have the `nbstripout` hook but this is in the cookie cutter, which I think is pretty critical to have if we are to be working with `.ipynb` directly. But if this can be enforced and we can encourage this rich diff then I agree `jupytext` is basically redundant. And just to flag to everyone more generally that we shouldn't be committing notebooks with any cell outputs.

**Comment by HarrisonWilde** [comment]
CC @sqr00t @Jack-Vines not sure who advocates for `jupytext` in the first place but would be good to get your inputs.

**Comment by HarrisonWilde** [comment]
Yes ok, so as I see it path to success looks like:

1. Subjective teaching to analysts needs to reinforce use and successful installation of pre commit hooks + emphasis on not committing any cell outputs in `ipynb`
2. Relaxing jupytext prescription (currently `*.ipynb` is in the `.gitignore`)
3. Those who want to can then still use it but we can safely allow for working entirely within `ipynb` especially for analysts who are also likely the ones most dependent on the cookie cutter?

**Comment by sqr00t** [comment]
On 1, agreed.

On 2, haven't gotten around to automate this. I suppose we could simply replace all pre commit config files with one that has nbstripout.

On 3, the effect of .gitignore should be that all .ipynb files are ignored except for those under a notebook folder.

On 4, if we do move towards monorepos, would be good to check that templates and subfolders will work with the same gitignore intention. I.e. .ipynb files only allowed under a notebooks folder.

**Comment by HarrisonWilde** [comment]
Thanks for the clarification on the `.gitignore` that part had slipped my mind. I suppose it depends how hard the prescription on no notebooks ever is in terms of analysis. I personally am happy with as few notebooks as possible but recognise realistically if we want analysts to work on simpler data requests etc. maybe sometimes a notebook is all that's needed / appropriate? That is a wider Q for sure, but for now I think let's close this as resolved as I think no real change is required to the repo? Is there somewhere we can keep track of an issue around automating / going back and replacing pre commit config files @sqr00t ?
